storage/storage-client/src/lib.rs

```diff
@@ -281,6 +282,52 @@ impl StorageRead for StorageReadServiceClient {
             .boxed();
         Ok(stream)
     }
+
+    // TODO migrate this to `ConfigStorage` trait as non-async
+    // and implement for LibraDBTrait once `StorageRead` is deprecated
+    async fn batch_fetch_config(&self, access_paths: Vec<AccessPath>) -> Vec<Option<Vec<u8>>> {
+        let mut requests = vec![];
+        // some access paths can have the same address, so don't duplicate requests for them
+
+        let mut req_addresses = vec![];
+        let mut seen_addresses = HashSet::new();
+        for path in access_paths.iter() {
+            let address = path.address;
+            if !seen_addresses.contains(&address) {
+                seen_addresses.insert(address.clone());
+                req_addresses.push(address.clone());
+                requests.push(RequestItem::GetAccountState { address });
+            }
+        }
+        let responses = match self.update_to_latest_ledger(0, requests).await {
+            Ok(responses) => responses.0,
+            Err(_) => vec![],
+        };
+        // Account address --> AccountState
+        let mut new_account_states = HashMap::new();
+
+        for (addr, resp) in req_addresses.into_iter().zip(responses) {
+            if let Ok(account_state) = resp.into_get_account_state_response() {
```

Comment:
not sure if it's intended, seems upon any query failure it'd return an empty vec?

Reply:
yes, this is intended